File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
arr = input().split('-')
ans = 0
if len(arr) == 1:
    print(sum(list(map(int, arr[0].split('+')))))
    exit()

ans += sum(list(map(int, arr[0].split('+'))))
for s in arr[1:]:
    if '+' in s:
        ans -= sum(list(map(int, s.split('+'))))
    else:
        ans -= int(s)
print(ans)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] s = br.readLine().split("-");
        int ans = 0;
        for (int i = 0; i < s.length; i++) {
            String ss = s[i];
            int tmp = 0;
            for (String num : ss.split("\\+")) {
                tmp += Integer.parseInt(num);
            }
            if (i == 0)
                ans += tmp;
            else
                ans -= tmp;
        }
        System.out.println(ans);
    }
}